[Feature]화면 흔들림 효과 추가

Kkumteulyi opened the pull request:
CamShake.Shake(흔들림 시간, 흔들림 정도,씬 카메라)를 사용하여 실행 가능합니다.
ex) 씬 카메라 -> scene.story = 스토리씬 카메라, scene.battle = 전투씬 카메라, scene.tutorial = 튜토리얼씬 카메라
## 개요
<!-- 어떤 점이 변경되었는지, 간단하게 작성해주세요. -->
CamShake.cs 코드 파일이 추가되었습니다.
## 변경 사항
<!-- 어떤 점에 변경되었는지 상세하게 작성해주세요.-->
### CamShake
- Shake(흔들림 시간, 흔들림 정도,카메라 번호) 함수를 호출하여 화면에 흔들림 효과 부여
- 각각의 씬에 사용되는 캔버스와 카메라를 가져와 관리하도록 **CamShakeManager**를 추가하였습니다.

### CamShakeManager(오브젝트)
- 스토리씬, 전투씬, 튜토리얼씬에서 사용되는 캔버스, 카메라를 가져온 뒤 개별적으로 흔들림 효과를 주기 위해 추가하였습니다.

*다른 함수에서 호출 시*
ex) CamShake.Instance.Shake(0.2f,5f,CamShake.scene.story)
## (참고 자료) 유니티 캔버스 렌더링 모드
### 월드 모드
- 3D 월드 좌표계를 기준으로 화면을 렌더링 함
- 캔버스 자체가 3D 오브젝트처럼 움직이므로 변형, 조작이 자유로워짐

### 스크린 모드
- 화면 좌표계를 기준으로 화면을 렌더링 함
- 스크린 모드의 경우 화면에 표시되는 요소들을 **고정적**으로 표시하므로 흔들림 효과를 주려면 월드 모드를 사용해야 함

Kkumteulyi committed:
화면 흔들림 효과 추가

CamShake.Shake(흔들림 시간, 흔들림 정도)를 사용하여 실행 가능합니다.

Kkumteulyi committed:
화면 흔들기 수정

튜토리얼씬, 스토리씬, 배틀씬 모두 적용가능하도록 구조를 변경하였습니다

miniron-v reviewed (changes requested):
고생하셨습니다! 화면이 잘 흔들리는 걸 코드와 함께 확인했는데, 사알짝 아쉬운 부분 + 간단한 개선점이 있어 리뷰 남깁니다.

## 1. 함수 인자로 정보를 요청할 땐
- Shake 함수의 마지막 인자는 canNum으로, **int형으로 카메라 구분**을 요청하고 있습니다. 하지만, **Shake 함수를 사용하는 사람 입장에서 이걸 알 방법은 없습니다.** 문서화를 해도 사용할 때마다 문서를 찾아야 하고, 호버링을 이용하더라도 매 번 함수 위에 마우스를 올리는 번거로움이 발생하죠.
- 이는 **enum**을 사용해 보다 직관적으로 해결할 수 있습니다.

## 2. 화면 흔들림 시 여백이 보이는 현상
- UI가 흔들리지 않는 것은 의도라고 여겨졌습니다만, 배경이 흔들림에 따라 **배경 바깥쪽(흰 바탕)이 보이는 현상**이 나타납니다.
- 이는 배경 캔버스의 **크기를 늘리거나**, 카메라의 **사이즈**를 조정하는 등의 방법으로 해결할 수 있습니다. (평소엔 보이지 않더라도, 흔들려서 움직인 만큼엔 배경이 이어져야 합니다.)

기능적으론 흠잡을 데가 없어 이 정도 수정만 완료되면 바로 머지하겠습니다.

Kkumteulyi committed:
화면 흔들림 효과 수정

1. 화면 흔들림 효과를 부여할 때 캔버스의 크기가 약간 확장된 뒤 효과가 나타나도록 수정
2. int형 매개 변수를 사용하는 대신 enum을 사용하여 직관성 개선
ex) scene.story , scene.battle , scene.tutorial

miniron-v reviewed (approved):
고생하셨습니다! 흔들림 잘 되는 거 확인했고 develop에 머지하겠습니다. 흔들리는 정도나 속도 등은 세부적으로 조정하면 될 것 같네요.

p.s. 유니티 enum 명명 규칙은 파스칼 케이스인데... 수정해주시겠죠 ㅎ..?

Kkumteulyi committed:
enum 대문자로 변경

enum 대문자로 수정
